doc/source/serve/key-concepts.md

```diff
@@ -30,6 +30,20 @@ handle = serve.run(my_first_deployment)
 print(ray.get(handle.remote())) # "Hello world!"
 ```
 
+(serve-key-concepts-application)=
+
+## Application
+
+An application is the unit of upgrade in a Ray Serve cluster. An application consists of one or more deployments. One of these deployments is considered the “ingress” deployment, which is where all inbound traffic is handled.
+
+Applications can be called via HTTP at the specified route_prefix or in Python by retrieving a handle to the application by name.
+
+(serve-key-concepts-cluster)=
+
+## Cluster
+
+A Serve cluster consists of all the Serve applications running on a single Ray cluster. A Serve application can talk to any other application running on the same cluster.
```

Comment:
I kinda think this one is unnecessary. Are there other places in docs we reference it?

Reply:
Not really, I can remove it. I think Angelina's suggestion of a diagram to explain the relationship here makes sense